source/adios2/engine/ssc/SscReader.cpp

```diff
@@ -117,6 +117,235 @@ void SscReader::SyncMpiPattern()
+            {
+                if (i == 0)
+                {
+                    m_WriterGlobalMpiInfo[i] = lrbuf;
+                    ++WriterInfoSize;
+                }
+                else
+                {
+                    int j_writersize;
+                    MPI_Recv( &j_writersize, 1, MPI_INT, AppStart[i], 96, MPI_COMM_WORLD, MPI_STATUS_IGNORE); //
+                    ++WriterInfoSize;
+
+                    m_WriterGlobalMpiInfo[i].resize(j_writersize);
+                    MPI_Recv( m_WriterGlobalMpiInfo[i].data(), j_writersize, MPI_INT, AppStart[i], 98, MPI_COMM_WORLD, MPI_STATUS_IGNORE); //
+                }
+            }
+
+            for (int i = m_WriterGlobalMpiInfo.size(); i < m_AppSize; ++i)
+            {
+                if (i == 0)
+                {
+                    m_ReaderGlobalMpiInfo[i] = lrbuf;
+                    ++ReaderInfoSize;
+                }
+                else
+                {
+                    int j_readersize;
+                    MPI_Recv( &j_readersize, 1, MPI_INT, AppStart[i], 95, MPI_COMM_WORLD, MPI_STATUS_IGNORE); //
+                    ++ReaderInfoSize;
+
+                    m_ReaderGlobalMpiInfo[i-m_WriterGlobalMpiInfo.size()].resize(j_readersize);
+                    MPI_Recv( m_ReaderGlobalMpiInfo[i-m_WriterGlobalMpiInfo.size()].data(), j_readersize, MPI_INT, AppStart[i], 97, MPI_COMM_WORLD, MPI_STATUS_IGNORE); //
+                }
+            }
+        }
+        else
+        {
+            MPI_Send(&m_ReaderSize, 1, MPI_INT, 0, 95, MPI_COMM_WORLD);
+            MPI_Send(lrbuf.data(), lrbuf.size(), MPI_INT, 0, 97, MPI_COMM_WORLD);
+        }
+    }
+
+
+    // Broadcast m_WriterGlobalMpiInfo and m_ReaderGlobalMpiInfo to all the processes.
+    MPI_Bcast(&WriterInfoSize, 1, MPI_INT, 0, MPI_COMM_WORLD); // Broadcast writerinfo size
+    MPI_Bcast(&ReaderInfoSize, 1, MPI_INT, 0, MPI_COMM_WORLD);
+    if (m_WorldRank != 0)
+    {
+        m_WriterGlobalMpiInfo.resize(WriterInfoSize);
+        m_ReaderGlobalMpiInfo.resize(ReaderInfoSize);
+    }
+
+    for (int i = 0; i < WriterInfoSize; ++i)
+    {
+        int ilen;
+        if (m_WorldRank == 0)
+        {
+            ilen = m_WriterGlobalMpiInfo[i].size();
+        }
+        MPI_Bcast(&ilen, 1, MPI_INT, 0, MPI_COMM_WORLD);
+        m_WriterGlobalMpiInfo[i].resize(ilen);
+        MPI_Bcast(m_WriterGlobalMpiInfo[i].data(), ilen, MPI_INT, 0, MPI_COMM_WORLD); // Broadcast readerinfo size
+    }
+
+    for (int i = 0; i < ReaderInfoSize; ++i)
+    {
+        int ilen;
+        if (m_WorldRank == 0)
+        {
+            ilen = m_ReaderGlobalMpiInfo[i].size();
+        }
+        MPI_Bcast(&ilen, 1, MPI_INT, 0, MPI_COMM_WORLD);
+        m_ReaderGlobalMpiInfo[i].resize(ilen);
+        MPI_Bcast(m_ReaderGlobalMpiInfo[i].data(), ilen, MPI_INT, 0, MPI_COMM_WORLD); // Broadcast readerinfo size
+    }
+
+
+    std::cout << "WorldRank " << m_WorldRank << std::endl;
```

Comment:
Could you remove these printing code when it's done?

Reply:
I have moved all of the printing codes to the end. They will be printed when m_Verbosity >= 10.